src/interfaces/token.ts

```diff
@@ -191,157 +187,76 @@ export class TokenInterface<C extends ChainId> extends ServiceInterface<C> {
   }
 
   /**
-   * Fetches supported zapper tokens and sets their icon
-   * @returns zapper tokens with icons
+   * Fetch the token amount that spender is allowed to spend on behalf of owner
+   * @param ownerAddress
+   * @param tokenAddress
+   * @param spenderAddress
+   * @returns TokenAllowance
    */
-  private async getZapperTokensWithIcons(): Promise<Token[]> {
-    const zapperTokens = await this.yearn.services.zapper.supportedTokens();
-
-    const zapperTokensAddresses = zapperTokens.map(({ address }) => address);
-
-    const zapperTokensIcons = await this.yearn.services.asset.ready.then(() =>
-      this.yearn.services.asset.icon(zapperTokensAddresses)
-    );
-
-    const setIcon = (token: Token): Token => {
-      const icon = zapperTokensIcons[token.address];
-      return icon ? { ...token, icon } : token;
-    };
-
-    return zapperTokens.map(setIcon);
-  }
-
-  /**
-   * Get allowance for a token either zapIn or direct deposit
-   * @param vault
-   * @param token
-   * @param amount
-   * @param account
-   * @returns transaction
-   */
-  async allowance(address: Address, vaultToken: Address, token: Address, account: Address): Promise<TokenAllowance> {
-    // If Ether is being sent, no need for approval
+  async allowance(ownerAddress: Address, tokenAddress: Address, spenderAddress: Address): Promise<TokenAllowance> {
     const allowance: TokenAllowance = {
-      owner: account,
-      spender: address,
-      amount: MaxUint256.toString(),
-      token
+      owner: ownerAddress,
+      token: tokenAddress,
+      spender: spenderAddress,
+      amount: MaxUint256.toString()
     };
 
-    if (EthAddress === token) return allowance;
-
-    if (vaultToken === token) {
-      const tokenContract = new Contract(token, TokenAbi, this.ctx.provider.read);
-      const partnerAddress = await this.yearn.services.partner?.address;
-      const addressToApprove = (this.shouldUsePartnerService(address) && partnerAddress) || address;
-      const allowanceAmount = await tokenContract.allowance(account, addressToApprove);
-
-      return {
-        ...allowance,
-        spender: addressToApprove,
-        amount: allowanceAmount.toString()
-      };
-    }
+    if (isNativeToken(tokenAddress) || tokenAddress === spenderAddress) return allowance;
 
-    const zapProtocol = PickleJars.includes(address) ? ZapProtocol.PICKLE : ZapProtocol.YEARN;
-
-    const zapInApprovalState = await this.yearn.services.zapper.zapInApprovalState(account, token, zapProtocol);
+    const tokenContract = new Contract(tokenAddress, TokenAbi, this.ctx.provider.read);
+    const allowanceAmount = await tokenContract.allowance(ownerAddress, spenderAddress);
 
     return {
-      owner: zapInApprovalState.ownerAddress,
-      spender: zapInApprovalState.spenderAddress,
-      amount: zapInApprovalState.allowance,
-      token
+      ...allowance,
+      amount: allowanceAmount.toString()
```

Comment:
Since we are using `ethers`, a BigNumber is mostly expected. Can see more here https://docs.ethers.io/v5/api/contract/contract/#Contract-functionsCall